### PR TITLE
adding the ability to save more than one image

### PR DESCRIPTION
**Pull request by linuxjuggler**
in the case of a file with multiple pages, this function will extract all the pages as images

am not sure about the testing cause at the end it will need to save the files somewhere ..

### CODE REVIEW

File: src/Pdf.php [hunk 1]
```diff
@@ -125,6 +125,31 @@ public function saveImage($pathToImage)
...
 
     /**
+     * Save the file as images to the given path.
+     *
+     * @param string $storeDirectory
+     * @param string $file_name     
+     *
+     * @return array $files the full path to the newly created images.
+     */
+    public function extractAllPagesAsImages($storeDirectory, $file_name)
```
Comment (freekmurze) [inline src/Pdf.php]: We use camelCasing for our variables. Could you please rename `$fileName`?

File: src/Pdf.php [hunk 2]
```diff
@@ -125,6 +125,31 @@ public function saveImage($pathToImage)
...
+    {
+        $pages = $this->getNumberOfPages();
+        $files = [];
+        for($i =1; $i <= $pages; $i++)
+        {
+            $this->setPage($i);
+            $file_path = "{$storeDirectory}/page_{$i}_{$file_name}.{$this->outputFormat}";
+            $imagick = $this->getImageData($file_path);
```
Comment (freekmurze) [inline src/Pdf.php]: `$imagick` contains the image data. It's better to call this variable `$imageData` 😄

**Comment by linuxjuggler** [comment]
done

File: src/Pdf.php [hunk 3]
```diff
@@ -125,6 +125,31 @@ public function saveImage($pathToImage)
...
+     * Save the file as images to the given path.
+     *
+     * @param string $storeDirectory
+     * @param string $file_name     
+     *
+     * @return array $files the full path to the newly created images.
+     */
+    public function extractAllPagesAsImages($storeDirectory, $file_name)
+    {
+        $pages = $this->getNumberOfPages();
```
Comment (freekmurze) [inline src/Pdf.php]: This variable does not contain the pages itself, but the number of page. Best to name it `$numberOfPages`

File: src/Pdf.php [hunk 4]
```diff
@@ -125,6 +125,31 @@ public function saveImage($pathToImage)
...
+        $files = [];
```
Comment (freekmurze) [inline src/Pdf.php]: If the number of pages is zero you can return early with and empty array.

File: src/Pdf.php [hunk 5]
```diff
@@ -125,6 +125,31 @@ public function saveImage($pathToImage)
...
+     * @param string $storeDirectory
+     * @param string $file_name     
+     *
+     * @return array $files the full path to the newly created images.
+     */
+    public function extractAllPagesAsImages($storeDirectory, $fileName)
+    {
+        $pages = $this->getNumberOfPages();
+        $files = [];
+        for($i =1; $i <= $pages; $i++)
```
Comment (freekmurze) [inline src/Pdf.php]: A for loop will do the job, but I think
`foreach(range(1,$numberOfPages) as $pageNumber) {` is more readable.

**Comment by freekmurze** [comment]
Hi, thanks for contributing! 👍 

I put a few remarks on the code. Could you please take a look at them.

**Comment by linuxjuggler** [comment]
@freekmurze am just testing the code locally before send the last commit

**Comment by linuxjuggler** [comment]
@freekmurze I think now everything is as you requested, I might missed something if so let me know.
thanks again

File: src/Pdf.php [hunk 6]
```diff
@@ -125,6 +125,32 @@ public function saveImage($pathToImage)
...
+     *
+     * @return array $files the full path to the newly created images.
+     */
+    public function extractAllPagesAsImages($storeDirectory, $fileName)
+    {
+        $numberOfPages = $this->getNumberOfPages();
+        if($numberOfPages === 0) return [];
+
+        $files = [];
+        foreach(range(1,$numberOfPages) as $pageNumber) {
```
Comment (freekmurze) [inline src/Pdf.php]: the foreach loop could be replaced by and `array_map`. In this way the `$files` will be not needed any more, improving readability even more.

**Comment by freekmurze** [comment]
One new remark added. If you're not familiar with `array_map`, no worries, Just let me know and I'll make the change myself after pulling.

**Comment by freekmurze** [comment]
On a sidenote: it seems that we use the same theme for our blogs 😃

**Comment by linuxjuggler** [comment]
@freekmurze I just sent another commit.

about the blog, actually I have modify it to suite arabic posts, cause some times I do need to write in my mother language.

thats way sadly I didn't upgrade it lately :(

**Comment by freekmurze** [comment]
Thanks again!

**Comment by linuxjuggler** [comment]
@freekmurze thank you for creating this simple nice package 👍

**Pull request merged**